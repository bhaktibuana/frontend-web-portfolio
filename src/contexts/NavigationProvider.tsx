import React, { ReactNode, useContext, useState } from "react";

interface ComponentProps {
  children: ReactNode;
}

type NavigationContextType = {
  homeNavOpen: boolean;
  setHomeNavOpen: (value: boolean) => void;
  adminNavOpen: boolean;
  setAdminNavOpen: (value: boolean) => void;
};

const NavigationContext = React.createContext<NavigationContextType | null>(
  null
);

export const useNavigation = () => {
  return useContext(NavigationContext) as NavigationContextType;
};

const NavigationProvider = ({ children }: ComponentProps) => {
  const [homeNavState, setHomeNavState] = useState<boolean>(false);
  const [adminNavState, setAdminNavState] = useState<boolean>(false);

  const setHomeNavOpen = (value: boolean): void => {
    setHomeNavState(value);
  };

  const setAdminNavOpen = (value: boolean): void => {
    setAdminNavState(value);
  };

  return (
    <>
      <NavigationContext.Provider
        value={{
          homeNavOpen: homeNavState,
          setHomeNavOpen,
          adminNavOpen: adminNavState,
          setAdminNavOpen,
        }}
      >
        {children}
      </NavigationContext.Provider>
    </>
  );
};

export default NavigationProvider;
