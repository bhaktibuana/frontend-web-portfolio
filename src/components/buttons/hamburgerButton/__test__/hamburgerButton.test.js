import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import HamburgerButton from "..";
import NavigationProvider from "../../../../contexts/NavigationProvider";

afterEach(() => {
  cleanup();
});

test("should render hamburger button", () => {
  render(
    <NavigationProvider>
      <HamburgerButton />
    </NavigationProvider>
  );
  const buttonElement = screen.getByTestId("hamburgerButton");

  expect(buttonElement).toBeInTheDocument();
  expect(buttonElement).toHaveClass("close");

  fireEvent.click(buttonElement);
  expect(buttonElement).toHaveClass("open");

  fireEvent.click(buttonElement);
  expect(buttonElement).toHaveClass("close");
});

test("matches snapshot", () => {
  const tree = renderer
    .create(
      <NavigationProvider>
        <HamburgerButton />
      </NavigationProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
