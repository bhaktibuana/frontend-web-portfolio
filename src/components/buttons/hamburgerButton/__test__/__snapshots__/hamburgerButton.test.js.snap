// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-beqWaB blyAlL close"
  color={
    Object {
      "bgDark": "#202020",
      "bgLight": "#ffffff",
      "primary": "#14b8a6",
      "secondary": "#1e293b",
      "tertiary": "#f1f5f9",
      "textDark": "#ffffff",
      "textLight": "#202020",
    }
  }
  data-testid="hamburgerButton"
  onClick={[Function]}
>
  <span />
  <span />
  <span />
  <span />
</div>
`;
